### DISCUSSION TITLE
Inpaint and Image Prompt does not work together

### DISCUSSION THREAD

**Discussion by willgodfather** [lllyasviel/Fooocus#2245] [General]
I want to use "Inpaint and "Image Prompt" together to do a face swap, but it is giving error every time I try to do it. 

I already enabled "Mixing Image Prompt and Inpaint" from Developer debug mode but still no luck. Does the app needs an update to fix it?

**Comment by mashb1t** [comment]
What's the error? Please provide the full terminal log.

**Reply by francopautasso1** [reply]
I have the same problem when performing 'Mixing Image Prompt and Inpaint' using Google Colab. The code is as follows:

[Parameters] Adaptive CFG = 7
[Parameters] Sharpness = 2
[Parameters] ADM Scale = 1.5 : 0.8 : 0.3
[Parameters] CFG = 3.0
[Parameters] Seed = 717459533649898409
[Fooocus] Downloading upscale models ...
[Fooocus] Downloading inpainter ...
[Inpaint] Current inpaint model is /content/Fooocus/models/inpaint/inpaint_v26.fooocus.patch
[Fooocus] Downloading control models ...
[Fooocus] Loading control models ...
extra clip vision: ['vision_model.embeddings.position_ids']
[Parameters] Sampler = dpmpp_2m_sde_gpu - karras
[Parameters] Steps = 30 - 15
[Fooocus] Initializing ...
[Fooocus] Loading models ...
Synthetic Refiner Activated
Synthetic Refiner Activated
Request to load LoRAs [['SDXL_FILM_PHOTOGRAPHY_STYLE_BetaV0.4.safetensors', 0.25], ['None', 1.0], ['None', 1.0], ['None', 1.0], ['None', 1.0], ('/content/Fooocus/models/inpaint/inpaint_v26.fooocus.patch', 1.0)] for model [/content/Fooocus/models/checkpoints/realisticStockPhoto_v20.safetensors].
Loaded LoRA [/content/Fooocus/models/loras/SDXL_FILM_PHOTOGRAPHY_STYLE_BetaV0.4.safetensors] for UNet [/content/Fooocus/models/checkpoints/realisticStockPhoto_v20.safetensors] with 788 keys at weight 0.25.
Loaded LoRA [/content/Fooocus/models/loras/SDXL_FILM_PHOTOGRAPHY_STYLE_BetaV0.4.safetensors] for CLIP [/content/Fooocus/models/checkpoints/realisticStockPhoto_v20.safetensors] with 264 keys at weight 0.25.
Loaded LoRA [/content/Fooocus/models/inpaint/inpaint_v26.fooocus.patch] for UNet [/content/Fooocus/models/checkpoints/realisticStockPhoto_v20.safetensors] with 960 keys at weight 1.0.
Request to load LoRAs [['SDXL_FILM_PHOTOGRAPHY_STYLE_BetaV0.4.safetensors', 0.25], ['None', 1.0], ['None', 1.0], ['None', 1.0], ['None', 1.0]] for model [/content/Fooocus/models/checkpoints/realisticStockPhoto_v20.safetensors].
Loaded LoRA [/content/Fooocus/models/loras/SDXL_FILM_PHOTOGRAPHY_STYLE_BetaV0.4.safetensors] for UNet [/content/Fooocus/models/checkpoints/realisticStockPhoto_v20.safetensors] with 788 keys at weight 0.25.
Requested to load SDXLClipModel
Loading 1 new model
unload clone 1
^C

**Reply by mashb1t** [reply]
> Note that this Colab will disable refiner by default because Colab free's resources are relatively limited (and some "big" features like image prompt may cause free-tier Colab to disconnect). We make sure that basic text-to-image is always working on free-tier Colab

see https://github.com/lllyasviel/Fooocus?tab=readme-ov-file#colab

I assume you're running into instance resource issues, either VRAM or RAM. Please check if adding these args after entry_with_update.py work for you: `--attention-split --disable-offload-from-vram --always-high-vram`

**Reply by poor7** [reply]
What does  `--attention-split`  give?

**Reply by mashb1t** [reply]
@poor7 earlier form of memory optimization due to cross attention, see https://github.com/AUTOMATIC1111/stable-diffusion-webui/wiki/Optimizations and https://github.com/basujindal/stable-diffusion/pull/117. Might also work without for you.

**Reply by poor7** [reply]
@mashb1t I tried it, the performance dropped by 2 times and the video memory consumption increased slightly. Interesting optimization...

**Reply by mashb1t** [reply]
@poor7 from experience, this totally depends on the used hardware. For Colab, this allows the usage of img2img as well as inpainting without it breaking and disconnecting. Feel free to use as you desire!
